### PR TITLE
initial commit of altdistsqlplan

### PR DESCRIPTION
**Pull request by sravotto**
Alternative distsql plan viewer.

The plan viewer uses the same encoding of the existing distsql plan viewer, but provides  different views: 
* Heatmap: a table that lists all the processors and edges with relative metrics
* Graph: A  simplified direct acyclic graph that shows how the various processors are used to execute the query. The graph only shows performance related information - additional information is available by hovering on nodes.
* SQL: The original SQL Query
* Plan: the plan in JSON format

To use, simply modify the url generated by `EXPLAIN ANALYZE (DISTSQL)`, replacing  `https://cockroachdb.github.io/distsqlplan/decode.html` with `https://cockroachdb.github.io/altdistsqlplan/decode.html`

### CODE REVIEW

File: altdistsqlplan/example.html [hunk 1]
```diff
@@ -0,0 +1 @@
...UiF43ItwXSosswo0IB__-9WcUTSGIomnSPoyiqetG0TRw508rP8PCjLGDexMdog51OsTc_eFXpNMRhIgxMlLFRiwyDZlORTazuf_8PTNJcaeQBJ6EiPqO77eK7f69hvm-91MLnaSH2MEjPkUjGAl5j_hwKGKuwdibLvqQ2_tcxzegkJjosQmKBtjB-Q3_7xR3XD46uNiaIao0HwAOyaNTH_U3g4GEAddCNtgy6B1T1J5MQJr48tWb7qerbu_8qntxfHzQIa9MKS5ODjrUPL3tXXTPZ88whXiyABXNQSxnRlYSK-X0FGb_Ht1wdbMS4eXjU950bd5PfkSRT9nPj4UjCzit54FzdnFydWTgYWZ1ClkC0pxzUIc2OqwCMi8o2msjaGxtI7AagEyyKkgq0eiK12LcaC8DsVLp6qi9ctTttVF7S1GT-qJBaonGvjTDiM-Yy1TlfowPZiz8DxVFU87gl82-SBRN-7RYYsSzBBEk9A3I9Xrk-952etRqO57nreoRoTvKUWtFjmZX1JOjDfVcYJz3cnLk1ZMjf49yRPYrR_7XlqO2W01sWgatWVOOaH1i07rE3icfDb8Vz9C15FmcqljM_QbQMr_XUdR_H0VTct2p59m97tgnjfb2k0bgVk0a_l4njcCtSe0NFV3o3uY3nzTKf5rPoTbdL7WDr01tWnfSIDtMGhW5noIai0xBaeKo9uwatCAZQIGuEhMZw0cp4vyaYtnLz-UbCShdWEmxOMoKkwlw8TApHyaLh-nSYZJHoyvnHcZY3nHVXCNzqhmz-jJEGkZjlKTqM5ooPjDURlVMdMvQ5xzcJoPW2gwy0HdCfkZDriGL70NUfOHMt-94qku5JaBApnyY_s4X24YVbTM_OJOmGNJbkzpdMM3laW5jTZqnPbePQBk0ll6p_b2wDSrtF0elaQXFtYDiboKE7tgo1MoTtgxJ-TCzk8y1s8yznm7aDzd3pGj9UnqetcFday2ZrcEDby8NzlaraUdlPe1XZmCPWoSL7Shcz89gHxS115UE1roSW13pRpbWqitZRaW1DUt3QsX3fQsqvl3OmxZUWl6wCRWv9qfDMiq-FZXALiDBiwsI8wrQ1kD6PfaZHZQtxga6DpRW0wYKJbuPDaz1UmODHZX1mlRW1Vly1aJKX3AabH9j-aDezs1O_f1Mg_k32uMP_wUAAP__K-8jbQ=="/>
```
Comment (RaduBerinde) [inline altdistsqlplan/example.html:1]: This should just be `url=decode.html#...`  so it works without a local web server.

File: altdistsqlplan/decode.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿<!DOCTYPE html>
+<head>
+  <script>
+  
+    location.replace('index.html' + window.location.hash);
+
```
Comment (RaduBerinde) [inline altdistsqlplan/decode.html:6]: [nit] blank lines

File: altdistsqlplan/css/style.css [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    transform: 
+      translate(60px, 0px)
+      rotate(315deg);
+    width: 40px;
+}
+.details th > div > span {
+    border-bottom: 1px solid #ccc;
+    padding: 5px 10px;
+}
+
```
Comment (RaduBerinde) [inline altdistsqlplan/css/style.css:99]: [nit] blank lines

File: altdistsqlplan/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+* Graph: A direct acyclic graph that shows how the various processors are used to execute the query
+* SQL: The original SQL Query
+* Plan: the plan in JSON format
+
+
+
+The viewer uses the same encoding of the viewer at `https://cockroachdb.github.io/distsqlplan/decode.html`.
+
+To use, simply modify the url generated by `EXPLAIN ANALYZE (DISTSQL)`, replacing  `https://cockroachdb.github.io/distsqlplan/decode.html` with `https://cockroachdb.github.io/altdistsqlplan/decode.html`
+
```
Comment (RaduBerinde) [inline altdistsqlplan/README.md:20]: [nit] a lot of extra blank lines in this file

File: altdistsqlplan/index.html [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    </link>
+    <link rel="stylesheet" href="css/w3theme.css"><!-- -->
+    </link>
+    <link rel="stylesheet" href="css/style.css"><!-- -->
+    </link>
+    <title>Alt DistSQL Plan Viewer</title>
+</head>
+
+
+
```
Comment (RaduBerinde) [inline altdistsqlplan/index.html:30]: [nit] too many blank lines

**Review by RaduBerinde — APPROVED** [review]
This is great, thanks! I plaid around with the example and it is really cool!

> To use, simply modify the url generated by EXPLAIN ANALYZE (DISTSQL), replacing https://cockroachdb.github.io/distsqlplan/decode.html with https://cockroachdb.github.io/altdistsqlplan/decode.html

Unfortunately, going the other way doesn't work because the URL changes to `index.html`. Maybe we should move `index.html` to `decode.html` and add an `index.html` that only shows the textbox and then goes to `decode.html`

**Comment by sravotto** [comment]
Thanks for reviewing the PR.
 One alternative to go back to the original distsql plan viewer is to add a link in the UI, perhaps in the toolbar at the top.

**Comment by RaduBerinde** [comment]
Links are now live! [example](https://cockroachdb.github.io/altdistsqlplan/index.html#eJzMV-9u2zYQ_76nIPgpxdRZJGVL1ic3RbsFTewifwYUUxDQ0sURaosuSSfOguxJBuwp9gJ7lD3JQMlebFmmZcdp-8UQeeLx7nf3-_n0gNWXIQ7x2bvjd2_PUZzqewfx28GBhFvIJvAKvT_tnSCZJqDQz6e9i4_o8FP-GnZwJhLo8hEoHP6GCXYwxQ5m-NLBYyliUEpIY3rIXzxKpjh0HZxm44k225cOjoUEHD5gneoh4BCf8_4QToEnIBsudnACmqfD3H0eQWcs0xGX5u6zMc9UiF43ItwXSosswo0IB__-9WcUTSGIomnSPoyiqetG0TRw508rP8PCjLGDexMdog51OsTc_eFXpNMRhIgxMlLFRiwyDZlORTazuf_8PTNJcaeQBJ6EiPqO77eK7f69hvm-91MLnaSH2MEjPkUjGAl5j_hwKGKuwdibLvqQ2_tcxzegkJjosQmKBtjB-Q3_7xR3XD46uNiaIao0HwAOyaNTH_U3g4GEAddCNtgy6B1T1J5MQJr48tWb7qerbu_8qntxfHzQIa9MKS5ODjrUPL3tXXTPZ88whXiyABXNQSxnRlYSK-X0FGb_Ht1wdbMS4eXjU950bd5PfkSRT9nPj4UjCzit54FzdnFydWTgYWZ1ClkC0pxzUIc2OqwCMi8o2msjaGxtI7AagEyyKkgq0eiK12LcaC8DsVLp6qi9ctTttVF7S1GT-qJBaonGvjTDiM-Yy1TlfowPZiz8DxVFU87gl82-SBRN-7RYYsSzBBEk9A3I9Xrk-952etRqO57nreoRoTvKUWtFjmZX1JOjDfVcYJz3cnLk1ZMjf49yRPYrR_7XlqO2W01sWgatWVOOaH1i07rE3icfDb8Vz9C15FmcqljM_QbQMr_XUdR_H0VTct2p59m97tgnjfb2k0bgVk0a_l4njcCtSe0NFV3o3uY3nzTKf5rPoTbdL7WDr01tWnfSIDtMGhW5noIai0xBaeKo9uwatCAZQIGuEhMZw0cp4vyaYtnLz-UbCShdWEmxOMoKkwlw8TApHyaLh-nSYZJHoyvnHcZY3nHVXCNzqhmz-jJEGkZjlKTqM5ooPjDURlVMdMvQ5xzcJoPW2gwy0HdCfkZDriGL70NUfOHMt-94qku5JaBApnyY_s4X24YVbTM_OJOmGNJbkzpdMM3laW5jTZqnPbePQBk0ll6p_b2wDSrtF0elaQXFtYDiboKE7tgo1MoTtgxJ-TCzk8y1s8yznm7aDzd3pGj9UnqetcFday2ZrcEDby8NzlaraUdlPe1XZmCPWoSL7Shcz89gHxS115UE1roSW13pRpbWqitZRaW1DUt3QsX3fQsqvl3OmxZUWl6wCRWv9qfDMiq-FZXALiDBiwsI8wrQ1kD6PfaZHZQtxga6DpRW0wYKJbuPDaz1UmODHZX1mlRW1Vly1aJKX3AabH9j-aDezs1O_f1Mg_k32uMP_wUAAP__K-8jbQ==)

**Pull request merged**